Coding | Replace "awk" by "mawk"

MichaIng opened the pull request:
**Status**: Testing
- [x] Passed first careful review
- [x] Second review on fresh morning with coffee 😄 ☕️
- [x] Script test

**Reference**: https://github.com/Fourdee/DietPi/issues/2323

**Commit list/description**:
+ DietPi-Cron | Coding: Replace "awk" by "mawk"
+ DietPi-Cron | Coding: Merge "grep" and "mawk" into single command
... and the same for all other scripts 😃

MichaIng committed:
v6.20

+ DietPi-Cron | Coding: Replace "awk" by "mawk"
+ DietPi-Cron | Coding: Merge "grep" and "mawk" into single command

MichaIng committed:
META | dietpi-cloud-migration; Replace "awk" by "mawk"

MichaIng committed:
v6.20

+ DietPi-Justboom | Coding: Replace "awk" by "mawk"
+ DietPi-Justboom | Coding: Use direct exit code check where applicable, instead of indirect check
+ DietPi-Justboom | Coding: Merge "grep" and "awk" where applicable
+ DietPi-Justboom | Coding: Minor

MichaIng committed:
v6.20

+ DietPi-Explorer | Coding: Replace "awk" by "mawk"
+ DietPi-Explorer | Coding: Use direct exit code check where applicable, instead of indirect check
+ DietPi-Explorer | Coding: Merge "grep" and "awk" where applicable
+ DietPi-Explorer | Coding: Minor

MichaIng committed:
v6.20

+ DietPi-Cloudshell | Coding: Replace "awk" by "mawk"
+ DietPi-Cloudshell | Coding: Merge "grep" and "awk" where applicable
+ DietPi-Cloudshell | Coding: Minor

Fourdee reviewed (approved):
@MichaIng 

Very nice, great work 👍 

Might take me a while to change my old habits from grep + awk to ```mawk```, however, performance gain is worth it 😄 

Please merge when ready.

MichaIng commented:
@Fourdee 
> Might take me a while to change my old habits from grep + awk to mawk, however, performance gain is worth it 😄

Jep, for me the same, I guess. We need to take care that Debians `mawk` version does not support `[[:blank:]]` and such, so `grep -m1 '^[[:blank:]]*setting' file | awk` cannot be merged without defining the space and tab differently (old style). Would be great, if we could create a package each arch from source: https://github.com/Fourdee/DietPi/issues/2323#issuecomment-455394808
This brings `mawk` onto current POSIX standards and fixes some bugs. Other distros have switched to mawk 1.3.4 since longer, but Debian somehow doesn't want to...

_________

Will try to apply at least the awk/mawk switch for all scripts tonight, but the grep/mawk merge is more work and one has to be concentrated to not create a syntax error. Perhaps apply this step-by-step in future DietPi versions 😄.

MichaIng committed:
v6.20

+ DietPi-Benchmark | Coding: Replace "awk" by "mawk"
+ DietPi-Benchmark | Coding: Merge "grep" and "awk" where applicable
+ DietPi-Benchmark | Coding: Minor

MichaIng committed:
v6.20

+ DietPi-Explorer | Syntax fix

MichaIng committed:
v6.20

+ DietPi-Swapfile | Coding: Replace "awk" by "mawk"
+ DietPi-Swapfile | Coding: Merge "grep" and "awk" where applicable
+ DietPi-Swapfile | Coding: Minor

MichaIng committed:
v6.20

+ DietPi-Sync | Coding: Replace "awk" by "mawk"
+ DietPi-Sync | Coding: Merge "grep" and "awk" where applicable

MichaIng committed:
v6.20

+ DietPi-WiFi-Monitor | Coding: Replace "awk" by "mawk"
+ DietPi-WiFi-Monitor | Coding: Minor

MichaIng committed:
v6.20

+ DietPi-LED_Control | Coding: Replace "awk" by "mawk"
+ DietPi-LED_Control | Coding: Minor

MichaIng committed:
Update postboot

+ DietPi-PostBoot | Coding: Replace "awk" by "mawk"

MichaIng committed:
v6.20

+ DietPi-Drive_Manager | Coding: Replace "awk" by "mawk"
+ DietPi-Drive_Manager | Coding: Merge "grep" and "awk" where applicable
+ DietPi-Drive_Manager | Coding: When assigning a single variable or command substitution to another variable, no double quotes are required to preserve line breaks and spaces. Only when adding as command or command argument, this is required.
+ DietPi-Drive_Manager | Coding: Minor

MichaIng committed:
v6.20

+ DietPi-Config | Coding: Replace "awk" by "mawk"
+ DietPi-Config | Coding: Merge "grep" and "awk" where applicable
+ DietPi-Config | Coding: Minor

MichaIng committed:
Merge branch 'dev' into mawk

MichaIng committed:
v6.20

+ DietPi-Morsecode | Coding: Replace "awk" by "mawk"
+ DietPi-Morsecode | Coding: Use exit trap (from DietPi-Globals) to remove temp file
+ DietPi-Morsecode | Coding: Minor

MichaIng committed:
v6.20

+ DietPi-CPU_set | Coding: Replace "awk" by "mawk"
+ DietPi-CPU_set | Coding: Minor

MichaIng committed:
v6.20

+ DietPi-Backup | Coding: Replace "awk" by "mawk"
+ DietPi-Backup | Coding: Minor

MichaIng committed:
v6.20

+ DietPi-Cleaner | Coding: Replace "awk" by "mawk"
+ DietPi-Cleaner | Coding: Minor

MichaIng committed:
v6.20

+ DietPi-Set_userdata | Coding: Replace "awk" by "mawk"
+ DietPi-Set_userdata | Coding: Minor

MichaIng committed:
v6.20

+ DietPi-Services | Coding: Replace "awk" by "mawk"
+ DietPi-Services | Coding: Minor

MichaIng committed:
v6.20

+ DietPi-Obtain_HW_model | Coding: Replace "awk" by "mawk"

MichaIng committed:
v6.20

+ DietPi-Set_software | Coding: Replace "awk" by "mawk"
+ DietPi-Set_software | Coding: Minor

MichaIng committed:
v6.20

+ DietPi-PREP | Coding: Replace "awk" by "mawk"
+ DietPi-PREP | Coding: Merge "grep" and "mawk" where applicable

MichaIng committed:
v6.20

+ DietPi-Globals | Coding: Replace "awk" by "mawk"

MichaIng committed:
v6.20

+ DietPi-Set_Hardware | Coding: Replace "awk" by "mawk"
+ DietPi-Set_Hardware | Coding: Minor

MichaIng committed:
v6.20

+ DietPi-Drive_Manager | Coding: (m)awk /pattern/ does not allow variables. Use -v pat="$var" '$0~pat' to achieve this instead.

MichaIng committed:
v6.20

+ DietPi-Explorer | Coding: (m)awk /pattern/ does not allow variables. Use -v pat="$var" '$0~pat' to achieve this instead.

MichaIng committed:
v6.20

+ DietPi-Backup | Coding: Syntax and minor

MichaIng committed:
v6.20

+ DietPi-Cleaner | Minor coding

MichaIng committed:
v6.20

+ DietPi-LED_control | Minor coding

MichaIng committed:
v6.20

+ DietPi-Set_userdata | Whoopsie

MichaIng commented:
Passed testing, will merge.
